### PR TITLE
mgmt, local generation for Compute ComputeRP 2024-07-01

### PR DESCRIPTION
**Pull request by v-hongli1**
- https://github.com/Azure/sdk-release-request/issues/5382

Manual codes:
sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/SharedGalleryImageTests.java
sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/VirtualMachineBootDiagnosticsTests.java
sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/VirtualMachineCustomImageOperationsTest.java
sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/VirtualMachineEMSILMSIOperationsTests.java
sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/VirtualMachineEncryptionOperationsTests.java
sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/VirtualMachineExtensionOperationsTests.java
sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/VirtualMachineManagedDiskOperationsTests.java
sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/VirtualMachineManagedServiceIdentityOperationsTests.java
sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/VirtualMachineOperationsTests.java
sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/VirtualMachinePopularImageTests.java
sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/VirtualMachineScaleSetEMSILMSIOperationsTests.java
sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/VirtualMachineScaleSetManagedDiskOperationsTests.java
sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/VirtualMachineScaleSetOperationsTests.java
sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/implementation/VirtualMachineUpdateTests.java

# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/VirtualMachinePopularImageTests.java [hunk 1]
```diff
@@ -37,34 +37,36 @@ public void canCreateAllPopularImageVM() {
...
+                .withSize(VirtualMachineSizeTypes.STANDARD_D2S_V3)
                 .createAsync();
             vmMonos.add(mono);
         }
 
         for (KnownLinuxVirtualMachineImage image : Arrays.stream(KnownLinuxVirtualMachineImage.values())
-            .filter(image -> image != KnownLinuxVirtualMachineImage.OPENSUSE_LEAP_15_1 && image != KnownLinuxVirtualMachineImage.SLES_15_SP1)
+            .filter(image -> image != KnownLinuxVirtualMachineImage.OPENSUSE_LEAP_15_1
+                && image != KnownLinuxVirtualMachineImage.SLES_15_SP1
+                && image != KnownLinuxVirtualMachineImage.ORACLE_LINUX_8_1)
```
Comment (XiaofeiCao) [inline sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/VirtualMachinePopularImageTests.java:56]: Is there a new version for oracle linux?

We may want to add deprecation message to `ORACLE_LINUX_8_1`, and add a newer version of it.

File: sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/VirtualMachineScaleSetEMSILMSIOperationsTests.java [hunk 2]
```diff
@@ -490,22 +490,24 @@ public void canUpdateVirtualMachineScaleSetWithEMSIAndLMSI() throws Exception {
...
                 .withExistingResourceGroup(virtualMachineScaleSet.resourceGroupName())
-                .withAccessToCurrentResourceGroup(BuiltInRole.CONTRIBUTOR);
+                .withAccessToCurrentResourceGroup(BuiltInRole.CONTRIBUTOR)
+                .create();
 
         // Update virtual machine so that it depends on the EMSI
         //
         virtualMachineScaleSet =
-            virtualMachineScaleSet.update().withNewUserAssignedManagedServiceIdentity(creatableIdentity).apply();
+            virtualMachineScaleSet.update().withExistingUserAssignedManagedServiceIdentity(creatableIdentity).apply();
```
Comment (XiaofeiCao) [inline sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/VirtualMachineScaleSetEMSILMSIOperationsTests.java:505]: Why this change?

File: sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/SharedGalleryImageTests.java [hunk 3]
```diff
@@ -35,18 +35,19 @@
...
     protected void initializeClients(HttpPipeline httpPipeline, AzureProfile profile) {
         rgName = generateRandomResourceName("javacsmrg", 15);
+        vmName = generateRandomResourceName("javavm", 12);
         super.initializeClients(httpPipeline, profile);
     }
 
     @Override
     protected void cleanUpResources() {
-        resourceManager.resourceGroups().beginDeleteByName(rgName);
+        resourceManager.resourceGroups().deleteByName(rgName);
```
Comment (XiaofeiCao) [inline sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/SharedGalleryImageTests.java:50]: why need to wait for delete completion? this may take longer for the test to finish

File: sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/implementation/VirtualMachineUpdateTests.java [hunk 4]
```diff
@@ -37,7 +38,9 @@ protected void cleanUpResources() {
     }
 
     @Test
+    @DoNotRecord(skipInPlayback = true)
     public void testVirtualMachineUpdate() {
+        // Management Long running operation is Failed or Cancelled
```
Comment (XiaofeiCao) [inline sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/implementation/VirtualMachineUpdateTests.java:43]: cause?

Comment (v-hongli1) [reply sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/implementation/VirtualMachineUpdateTests.java:43]: Regardless of `live` mode or `record` mode, `Long running operation is Failed or Cancelled` Exception occurs. The detail information following as:
[test.log](https://github.com/user-attachments/files/16689008/test.log)
[compute-test.log](https://github.com/user-attachments/files/16689016/compute-test.log)

File: sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/VirtualMachinePopularImageTests.java (hunk 1 above)
Comment (v-hongli1) [reply sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/VirtualMachinePopularImageTests.java:56]: The `ORACLE_LINUX_8_1` seems to be deprecated.
The detail information following as:
[test.log](https://github.com/user-attachments/files/16689115/test.log)
[compute-test.log](https://github.com/user-attachments/files/16689127/compute-test.log)

**Comment by weidongxu-microsoft** [comment]
> * [[resource manager] Java: Release for Compute ComputeRP 2024-07-01 sdk-release-request#5382](https://github.com/Azure/sdk-release-request/issues/5382)
> 
> Manual codes: sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/SharedGalleryImageTests.java sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/VirtualMachineBootDiagnosticsTests.java sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/VirtualMachineCustomImageOperationsTest.java sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/VirtualMachineEMSILMSIOperationsTests.java sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/VirtualMachineEncryptionOperationsTests.java sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/VirtualMachineExtensionOperationsTests.java sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/VirtualMachineManagedDiskOperationsTests.java sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/VirtualMachineManagedServiceIdentityOperationsTests.java sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/VirtualMachineOperationsTests.java sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/VirtualMachinePopularImageTests.java sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/VirtualMachineScaleSetEMSILMSIOperationsTests.java sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/VirtualMachineScaleSetManagedDiskOperationsTests.java sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/VirtualMachineScaleSetOperationsTests.java sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/implementation/VirtualMachineUpdateTests.java

Use a commit for manual code, instead of list the file. This is not readable.

File: sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/implementation/VirtualMachineUpdateTests.java (hunk 4 above)
Comment (weidongxu-microsoft) [reply sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/implementation/VirtualMachineUpdateTests.java:43]: `testVirtualMachineUpdate` is the basic test case. It MUST pass.

File: sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/VirtualMachineEMSILMSIOperationsTests.java [hunk 5]
```diff
@@ -428,7 +428,8 @@ public void canUpdateVirtualMachineWithEMSIAndLMSI() throws Exception {
 
         // Ensure the "User Assigned (External) MSI" id can be retrieved from the virtual machine
         //
-        Set<String> emsiIds = virtualMachine.userAssignedManagedServiceIdentityIds();
+        Set<String> emsiIds = virtualMachine.innerModel().identity().userAssignedIdentities().keySet();
+        emsiIds.removeIf(emsiId -> !emsiId.endsWith("/" + identityName1));
```
[the remark this answers is not in the document]
Comment (v-hongli1) [reply sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/VirtualMachineEMSILMSIOperationsTests.java:432]: n live mode and Record mode testing, invalid identity data is mixed into the identity, but surprisingly, this invalid data is not recorded in the Record session.
The detail information following as test.log:2193
[test.log](https://github.com/user-attachments/files/16690083/test.log)

**Comment by weidongxu-microsoft** [comment]
Also, for the failing test that you are modifying, does them work on main branch (without api-version update)?

File: sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/VirtualMachineEMSILMSIOperationsTests.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (XiaofeiCao) [reply sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/VirtualMachineEMSILMSIOperationsTests.java:432]: @v-hongli1 Please don't directly paste the log here, as it contains raw sensitive information, e.g. key, secret, etc. Only paste necessary redacted information.

File: sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/implementation/VirtualMachineUpdateTests.java (hunk 4 above)
Comment (XiaofeiCao) [reply sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/implementation/VirtualMachineUpdateTests.java:43]: @v-hongli1 Please remove the log link in the comment and investigate the LRO failure cause.

File: sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/VirtualMachineEMSILMSIOperationsTests.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (XiaofeiCao) [reply sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/VirtualMachineEMSILMSIOperationsTests.java:432]: Seems azure now will always add a fixed user-assigned identity to the VM..
```json
"userAssignedIdentities": {
 
"/subscriptions/xx/resourceGroups/AzSecPackAutoConfigRG/providers/Microsoft.ManagedIdentity/userAssignedIdentities/AzSecPackAutoConfigUA-westus2": {}
}

```

@weidongxu-microsoft Are you aware of this behavior? I guess it's due to security reason, and we have to adopt it..

[the remark this answers is not in the document]
Comment (weidongxu-microsoft) [reply sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/VirtualMachineEMSILMSIOperationsTests.java:432]: Yeah, my assumption is this is a security feature applies to the subscription the test runs on. We may need to adjust the test case.

[the remark this answers is not in the document]
Comment (XiaofeiCao) [reply sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/VirtualMachineEMSILMSIOperationsTests.java:432]: @v-hongli1 Can we adjust the assertion to still use `virtualMachine.userAssignedManagedServiceIdentityIds()` and verify it contains the identity id?

File: sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/implementation/VirtualMachineUpdateTests.java (hunk 4 above)
Comment (weidongxu-microsoft) [reply sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/implementation/VirtualMachineUpdateTests.java:43]: Same comment as Xiaofei. What we need from you is to either fix the issue (e.g. if a simple update to different region/sku), or at least point to place in code where it fails. Do not paste the log.

Comment (weidongxu-microsoft) [reply sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/implementation/VirtualMachineUpdateTests.java:43]: WIP https://github.com/Azure/azure-sdk-for-java/pull/41596

File: sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/VirtualMachineEMSILMSIOperationsTests.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (weidongxu-microsoft) [reply sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/VirtualMachineEMSILMSIOperationsTests.java:432]: During my test run, I still can see this
```
  "identity" : {
    "type" : "SystemAssigned"
  }
```

File: sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/implementation/VirtualMachineUpdateTests.java (hunk 4 above)
Comment (v-hongli1) [reply sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/implementation/VirtualMachineUpdateTests.java:42]: Fixed it with WIP https://github.com/Azure/azure-sdk-for-java/pull/41596

File: sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/SharedGalleryImageTests.java (hunk 3 above)
Comment (v-hongli1) [reply sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/SharedGalleryImageTests.java]: Fixed in the new version.

File: sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/VirtualMachineEMSILMSIOperationsTests.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (v-hongli1) [reply sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/VirtualMachineEMSILMSIOperationsTests.java:432]: > @v-hongli1 Can we adjust the assertion to still use `virtualMachine.userAssignedManagedServiceIdentityIds()` and verify it contains the identity id?

Yes, but it's a little troublesome for `Creatable<Identity>`

File: sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/VirtualMachineScaleSetEMSILMSIOperationsTests.java (hunk 2 above)
Comment (v-hongli1) [reply sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/VirtualMachineScaleSetEMSILMSIOperationsTests.java]: This change has been rollbacked.

File: sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/VirtualMachineEMSILMSIOperationsTests.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (XiaofeiCao) [reply sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/VirtualMachineEMSILMSIOperationsTests.java:432]: We have the identity name. We could either do
1. iterate the identity ids, verify one contains the identity name
2. get the Identity by identity name, via `MsiManager.identities().getByResourceGroup()`, then a contains check.

File: sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/implementation/VirtualMachineUpdateTests.java [hunk 6]
```diff
@@ -38,13 +38,12 @@ protected void cleanUpResources() {
...
 
     @Test
-    @DoNotRecord(skipInPlayback = true)
     public void testVirtualMachineUpdate() {
         // Management Long running operation is Failed or Cancelled
         final String vmname = "javavm1";
 
-        final String mySqlInstallScript = "https://raw.githubusercontent.com/Azure/azure-quickstart-templates/4397e808d07df60ff3cdfd1ae40999f0130eb1b3/mysql-standalone-server-ubuntu/scripts/install_mysql_server_5.6.sh";
-        final String installCommand = "bash install_mysql_server_5.6.sh Abc.123x(";
+        final String mySqlInstallScript = "https://raw.githubusercontent.com/Azure/azure-sdk-for-java/0f875b569cb30b3cd4232bd20abb02452431b1ad/sdk/resourcemanager/azure-resourcemanager-compute/src/test/assets/install_mysql_server_5.7.sh";
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/implementation/VirtualMachineUpdateTests.java]: use commit id `00df5c3ae1e25c526e265e78a00211d068b94f93` in path (it is on main branch)

Comment (v-hongli1) [reply sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/implementation/VirtualMachineUpdateTests.java]: Fixed in the new version.

File: eng/code-quality-reports/src/main/resources/revapi/revapi.json [hunk 7]
```diff
@@ -633,6 +633,12 @@
           "old": "class com.azure.resourcemanager.sql.models.UserIdentity",
           "new": "class com.azure.resourcemanager.sql.models.UserIdentity",
           "justification": "Customer unlikely to subclass this class."
+        },
+        {
+          "ignore": true,
+          "code": "java.method.removed",
+          "old": "method com.azure.resourcemanager.compute.models.WindowsConfiguration com.azure.resourcemanager.compute.models.WindowsConfiguration::withEnableVMAgentPlatformUpdates(java.lang.Boolean)",
+          "justification": "Fix method removed."
```
Comment (XiaofeiCao) [inline eng/code-quality-reports/src/main/resources/revapi/revapi.json:641]: `Service changed the property to readOnly.`

https://github.com/Azure/azure-rest-api-specs/pull/29575#discussion_r1653671646
It's been approved.

Also mention it in `Breaking Changes` section.

**Review by XiaofeiCao — APPROVED** [review]

File: eng/code-quality-reports/src/main/resources/revapi/revapi.json (hunk 7 above)
Comment (v-hongli1) [reply eng/code-quality-reports/src/main/resources/revapi/revapi.json]: Fixed in the new version.
